tests/api/retrieval.test.js

```diff
@@ -0,0 +1,312 @@
+import { createMocks } from "node-mocks-http";
+import retrieval from "../../pages/api/retrieval";
+import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
+import executeQuery from "@lib/integration/queryManager";
+import jwt from "jsonwebtoken";
+
+jest.mock("next-auth/client");
+jest.mock("@aws-sdk/client-dynamodb");
+jest.mock("@lib/integration/queryManager");
+
+jest.mock("@lib/integration/dbConnector", () => {
+  const mockClient = {
+    connect: jest.fn(),
+    query: jest.fn(),
+    end: jest.fn(),
+  };
+  return jest.fn(() => mockClient);
+});
+
+const dynamoClient = {
+  send: jest.fn(),
+};
+
+describe("/api/retrieval", () => {
+  beforeAll(() => {
+    process.env.TOKEN_SECRET = "gc-form-super-secret-code";
+  });
+  afterAll(() => {
+    delete process.env.TOKEN_SECRET;
+  });
+
+  it("Should return 400 status code b/c formID is an empty string", async () => {
+    const token = jwt.sign(
+      {
+        formID: 1,
+      },
+      process.env.TOKEN_SECRET,
+      {
+        expiresIn: "1y",
+      }
+    );
+
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Origin: "http://localhost:3000",
+        authorization: `Bearer ${token}`,
+      },
+      query: {
+        maxRecords: "10",
+        formID: "",
+      },
+    });
+    await retrieval(req, res);
+    expect(res.statusCode).toBe(400);
+    expect(JSON.parse(res._getData())).toEqual(expect.objectContaining({ error: "Bad Request" }));
+  });
+
+  it("Should return 400 status code b/c formID is undefined", async () => {
+    const token = jwt.sign(
+      {
+        formID: 1,
+      },
+      process.env.TOKEN_SECRET,
+      {
+        expiresIn: "1y",
+      }
+    );
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Origin: "http://localhost:3000",
+        authorization: `Bearer ${token}`,
+      },
+      query: {
+        maxRecords: "10",
+        formID: undefined,
+      },
+    });
+    await retrieval(req, res);
+    expect(res.statusCode).toBe(400);
+    expect(JSON.parse(res._getData())).toEqual(expect.objectContaining({ error: "Bad Request" }));
+  });
+
+  it("Should return 400 status when maxRecord is out of the range 1-10", async () => {
+    const token = jwt.sign(
+      {
+        email: "forms@cds.ca",
+      },
+      process.env.TOKEN_SECRET,
+      {
+        expiresIn: "1y",
+      }
+    );
+    const { req, res } = createMocks({
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Origin: "http://localhost:3000/api/retrieval?maxRecords=1",
+        authorization: `Bearer ${token}`,
+      },
+      query: {
+        maxRecords: "11",
+        formID: "22",
+      },
+    });
+    executeQuery.mockImplementation((client, sql) => {
```

Comment:
This mockImplementation for this query should never be called in this test, because of the early return and 400 error.